tests/e2e/util.go

```diff
@@ -4182,3 +4184,80 @@ func isCsiFssEnabled(ctx context.Context, client clientset.Interface, namespace
 		gomega.BeTrue(), "FSS %s not found in the %s configmap in namespace %s", fss, csiFssCM, namespace)
 	return false
 }
+
+// getVolumeSnapshotClassSpec returns a spec for the volume snapshot class
+func getVolumeSnapshotClassSpec(deletionPolicy snapc.DeletionPolicy,
+	parameters map[string]string) *snapc.VolumeSnapshotClass {
+	var volumesnapshotclass = &snapc.VolumeSnapshotClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "VolumeSnapshotClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "volumesnapshot-",
+		},
+		Driver:         "csi.vsphere.vmware.com",
+		DeletionPolicy: deletionPolicy,
+	}
+
+	if parameters != nil {
+		volumesnapshotclass.Parameters = parameters
+	}
+	return volumesnapshotclass
+}
+
+// getVolumeSnapshotSpec returns a spec for the volume snapshot
+func getVolumeSnapshotSpec(namespace string, snapshotclassname string, pvcName string) *snapc.VolumeSnapshot {
+	var volumesnapshotSpec = &snapc.VolumeSnapshot{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "VolumeSnapshot",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "snapshot-",
+			Namespace:    namespace,
+		},
+		Spec: snapc.VolumeSnapshotSpec{
+			VolumeSnapshotClassName: &snapshotclassname,
+			Source: snapc.VolumeSnapshotSource{
+				PersistentVolumeClaimName: &pvcName,
+			},
+		},
+	}
+	return volumesnapshotSpec
+}
+
+// waitForVolumeSnapshotReadyToUse waits for the volume's snapshot to be in ReadyToUse
+func waitForVolumeSnapshotReadyToUse(client snapclient.Clientset, ctx context.Context, namespace string,
+	name string) (*snapc.VolumeSnapshot, error) {
+	var volumeSnapshot *snapc.VolumeSnapshot
+	var err error
+	waitErr := wait.PollImmediate(resizePollInterval, totalResizeWaitPeriod, func() (bool, error) {
+		volumeSnapshot, err = client.SnapshotV1().VolumeSnapshots(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error fetching volumesnapshot details : %v", err)
+		}
+		if volumeSnapshot.Status != nil && *volumeSnapshot.Status.ReadyToUse {
+			return true, nil
+		}
+		return false, nil
+	})
+	return volumeSnapshot, waitErr
+}
+
+// waitForVolumeSnapshotContentToBeDeleted wait till the volume snapshot content is deleted
+func waitForVolumeSnapshotContentToBeDeleted(client snapclient.Clientset, ctx context.Context,
+	name string) (*snapc.VolumeSnapshotContent, error) {
+	var volumeSnapshotContent *snapc.VolumeSnapshotContent
+	var err error
+	waitErr := wait.PollImmediate(resizePollInterval, totalResizeWaitPeriod, func() (bool, error) {
```

Comment:
Use different variables for poll interval and wait period, or change the variable name to make it generic.

Reply:
Changed to generic variables